tests/Benchmarks/System.Text.Formatting/StringFormatterPerf.cs

```diff
@@ -0,0 +1,241 @@
+            var sb = new StringFormatter(NumbersToWrite, pool);
+            var format = new StandardFormat('X', StandardFormat.NoPrecision);
+
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)), format);
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string InvariantFormatIntHexClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)).ToString("X"));
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string InvariantFormatStruct()
+        {
+            var sb = new StringFormatter(NumbersToWrite * 2, pool);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(new Age(i % 10));
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string FormatGuid()
+        {
+            var guid = Guid.NewGuid();
+            var guidsToWrite = NumbersToWrite / 10;
+
+            var sb = new StringFormatter(guidsToWrite * 36, pool);
+            for (int i = 0; i < guidsToWrite; i++)
+            {
+                sb.Append(guid);
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string InvariantFormatStructClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite * 2);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(new Age(i % 10));
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string CustomCultureFormat()
+        {
+            var sb = new StringFormatter(NumbersToWrite * 3, pool);
+            sb.SymbolTable = _symbolTable;
+
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                var next = (i % 128) + 101;
+                sb.Append(next);
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string CustomCultureFormatClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite * 3);
+            var culture = new CultureInfo("th");
+
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((i % 128) + 100).ToString(culture));
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public void EncodeStringToUtf8()
+        {
+            const string text = "Hello World!";
+            int stringsToWrite = 2000;
+            int size = stringsToWrite * text.Length + stringsToWrite;
+            var formatter = new ArrayFormatter(size, SymbolTable.InvariantUtf8, pool);
+
+            for (int i = 0; i < stringsToWrite; i++)
+            {
+                formatter.Append(text);
+                formatter.Append(1);
+            }
```

Comment:
can formater return something here? to compare apples to apples with `EncodeStringToUtf8Clr` which does `Encoding.UTF8.GetBytes(formatter.ToString())`?

Reply:
There is nothing to return from the array formatter, unless you want to return a span.
 https://github.com/dotnet/corefxlab/blob/master/src/System.Text.Formatting/System/Text/Formatting/Formatters/ArrayFormatter.cs#L43

Reply:
You can return the arraysegment: `formatter.Formatted`